python/lsst/ap/association/diaPipe.py

```diff
@@ -0,0 +1,228 @@
+                        pipelineConnections=DiaPipelineConnections):
+    """Config for DiaPipelineTask.
+    """
+    apdb = pexConfig.ConfigurableField(
+        target=daxApdb.Apdb,
+        ConfigClass=daxApdb.ApdbConfig,
+        doc="Database connection for storing associated DiaSources and "
+            "DiaObjects. Must already be initialized.",
+    )
+    diaSourceDpddifier = pexConfig.ConfigurableField(
+        target=MapDiaSourceTask,
+        doc="Task for assigning columns from the raw output of ip_diffim into "
+            "a schema that more closely resembles the DPDD.",
+    )
+    diaCatalogLoader = pexConfig.ConfigurableField(
+        target=LoadDiaCatalogsTask,
+        doc="Task to load DiaObjects and DiaSources from the Apdb.",
+    )
+    associator = pexConfig.ConfigurableField(
+        target=AssociationTask,
+        doc="Task used to associate DiaSources with DiaObjects.",
+    )
+    diaForcedSource = pexConfig.ConfigurableField(
+        target=DiaForcedSourceTask,
+        doc="Task used for force photometer DiaObject locations in direct and "
+            "difference images.",
+    )
+
+    def setDefaults(self):
+        self.apdb.dia_object_index = "baseline"
+        self.apdb.dia_object_columns = []
+        self.apdb.extra_schema_file = os.path.join(
+            getPackageDir("ap_association"),
+            "data",
+            "apdb-ap-pipe-schema-extra.yaml")
+
+    def validate(self):
+        pexConfig.Config.validate(self)
+        if self.diaCatalogLoader.htmLevel != \
+           self.associator.diaCalculation.plugins["ap_HTMIndex"].htmLevel:
+            raise ValueError("HTM index level in LoadDiaCatalogsTask must be "
+                             "equal to HTMIndexDiaCalculationPlugin index "
+                             "level.")
+
+
+class DiaPipelineTask(pipeBase.PipelineTask):
+    """Task for loading, associating and storing Difference Image Analysis
+    (DIA) Objects and Sources.
+    """
+    ConfigClass = DiaPipelineConfig
+    _DefaultName = "diaPipe"
+    RunnerClass = pipeBase.ButlerInitializedTaskRunner
+
+    def __init__(self, initInputs=None, **kwargs):
+        super().__init__(**kwargs)
+        self.apdb = self.config.apdb.apply(
+            afw_schemas=dict(DiaObject=make_dia_object_schema(),
+                             DiaSource=make_dia_source_schema()))
+        self.makeSubtask("diaSourceDpddifier",
+                         inputSchema=initInputs["diaSourceSchema"])
+        self.makeSubtask("diaCatalogLoader")
+        self.makeSubtask("associator")
+        self.makeSubtask("diaForcedSource")
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        expId, expBits = butlerQC.quantum.dataId.pack("visit_detector",
+                                                      returnMaxBits=True)
+        inputs["ccdExposureIdBits"] = expBits
+
+        outputs = self.run(**inputs)
```

Comment:
Couldn't you just pass `ccdExposureIdBits = expBits` directly, rather than packing it in `inputs`?

Reply:
Mostly I did this parroting what was happening in calibration.py. They do a similar operation with the same value/variable.